observer/probers/tls/tls_conf.go

```diff
@@ -0,0 +1,137 @@
+package probers
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	certExpiryName = "obs_cert_expiry"
+	outcomeName    = "tls_prober_outcome"
+)
+
+// TLSConf is exported to receive YAML configuration.
+type TLSConf struct {
+	URL      string `yaml:"url"`
+	RootOrg  string `yaml:"rootOrg"`
+	RootCN   string `yaml:"rootCN"`
+	Response string `yaml:"response"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c TLSConf) Kind() string {
+	return "TLS"
+}
+
+// UnmarshalSettings takes YAML as bytes and unmarshals it to the to an TLSConf
+// object.
+func (c TLSConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf TLSConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c TLSConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got %q, expected a valid url: %s", c.URL, err)
+	}
+	if url.Scheme == "" {
```

Comment:
Looking at the example in [the docs for tls.Dial()](https://pkg.go.dev/crypto/tls#Dial), I'm wondering if we actually _want_ the scheme to be empty, rather than the other way around.  What do you think?

Reply:
Oh that's a really good point! Yeah I like that solution: don't allow people to specify a scheme, because we know we're going to be using tls anyway.